core/src/main/java/org/elasticsearch/cluster/routing/allocation/decider/EnableAllocationDecider.java

```diff
@@ -98,7 +98,7 @@ public void setEnableAllocation(Allocation enableAllocation) {
     @Override
     public Decision canAllocate(ShardRouting shardRouting, RoutingNode node, RoutingAllocation allocation) {
         if (allocation.ignoreDisable()) {
-            return allocation.decision(Decision.YES, NAME, "allocation is explicitly ignoring any disabling of allocation");
+            return allocation.decision(Decision.YES, NAME, "explicitly ignoring any disabling of allocation due to manual _reroute call");
```

Comment:
I could be wrong, but it seems the reroute API goes through this block: https://github.com/elastic/elasticsearch/blob/master/core/src/main/java/org/elasticsearch/cluster/routing/allocation/AllocationService.java#L256

Regardless of whether allocation commands are set or not (commands will be empty if none are set in the reroute API call).